src/gitlab/services/gitlab-api/pipelines-api.ts

```diff
@@ -58,17 +58,24 @@ export namespace GitlabPipelines {
         return result.promise;
     }
 
-    export function getPipelinesForMR(mergeRequestId: number) {
+    export async function getPipelinesForMR(mergeRequestId: number) {
         const projectId = getProjectId();
-        const result = new Deferred<PipelineBase[]>();
 
-        $.ajax(`/api/v4/projects/${projectId}/merge_requests/${mergeRequestId}/pipelines`)
-            .then((data: PipelineScheduleBase[]) => {
-                result.resolve(data);
-            })
-            .catch(result.reject);
-
-        return result.promise;
+        try {
+            const mrData = await $.ajax(`/api/v4/projects/${projectId}/merge_requests/${mergeRequestId}`);
```

Comment:
If not sure whether we should mix async and sync code. I would keep the standard in this project.

Reply:
It looks as if Monar is using the `Deffered` object just as a Promise factory - he awaits the jquery promise, then resolves the object manually, and then unwraps a new-and-just-now-resolved `Promise` from the `Deffered` :)  
So I just converted the function to async/await, that is a standard way, and there is no mixing of sync/async, because the whole thing is async.